### PR TITLE
Fix Bug 1454501 - apply dark theme only to builtin Dark theme

### PR DESCRIPTION
**Pull request by rlr**

### CODE REVIEW

File: system-addon/test/unit/lib/ThemeFeed.test.js [hunk 1]
```diff
@@ -49,32 +49,21 @@ describe("ThemeFeed", () => {
...
+      feed.updateTheme({id: "firefox-compact-dark@mozilla.org"});
       assert.calledOnce(dispatch);
       assert.equal(dispatch.firstCall.args[0].data.className, "dark-theme");
     });
-    it("should switch to default theme if the current theme has dark text", () => {
+    it("should switch to default theme if the current theme is not the builtin Dark theme", () => {
       const dispatch = sinon.spy();
       feed.store.dispatch = dispatch;
-      feed.updateTheme({textcolor: "#111"});
+      feed.updateTheme({textcolor: "firefox-compact-light@mozilla.org"});
```
Comment (Mardak) [inline system-addon/test/unit/lib/ThemeFeed.test.js]: probably want `id` here

File: system-addon/test/unit/lib/ThemeFeed.test.js [hunk 2]
```diff
@@ -49,32 +49,21 @@ describe("ThemeFeed", () => {
...
       dispatch.reset();
-      feed.updateTheme({textcolor: "brown"});
-      assert.calledOnce(dispatch);
-      assert.equal(dispatch.firstCall.args[0].data.className, "");
-    });
-    it("should switch to default theme by default (missing textcolor on theme)", () => {
-      const dispatch = sinon.spy();
-      feed.store.dispatch = dispatch;
-      feed.updateTheme();
+      feed.updateTheme({textcolor: undefined});
```
Comment (Mardak) [inline system-addon/test/unit/lib/ThemeFeed.test.js]: `id` here too

File: system-addon/lib/ThemeFeed.jsm [hunk 3]
```diff
@@ -26,28 +26,15 @@ this.ThemeFeed = class ThemeFeed {
...
-    // text. If light, then we enable our dark (background) theme.
-    const textcolor = (data && data.textcolor) || "black";
-    const window = Services.wm.getMostRecentWindow("navigator:browser");
-    const dummy = window.document.createElement("dummy");
-    dummy.style.color = textcolor;
-    const [r, g, b] = this._parseRGB(window.getComputedStyle(dummy).color);
-    const luminance = 0.2125 * r + 0.7154 * g + 0.0721 * b;
-    const className = (luminance <= 110) ? "" : "dark-theme";
+    // If the theme is the built-in Dark theme, then activate our dark theme.
+    const className = (data.id === "firefox-compact-dark@mozilla.org") ? "dark-theme" : "";
```
Comment (Mardak) [inline system-addon/lib/ThemeFeed.jsm]: nit: parens aren't needed

File: system-addon/test/unit/lib/ThemeFeed.test.js (hunk 2 above)
Comment (rlr) [reply system-addon/test/unit/lib/ThemeFeed.test.js]: indeed. derp

**Comment by rlr** [comment]
Hmmm, I am noticing that when going from Dark -> Default theme we are staying stuck with a dark NTP page. looking into it

**Comment by rlr** [comment]
AHA! `data` is null because Default theme is no theme at all.

screen shot 2018-04-17 at 12 44 22 pm

**Pull request merged**